Add epsilon argument to Adagrad optimizer

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24980 Add epsilon argument to Adagrad optimizer**

We'll need this internally, so just updating the open source version. the other optimizers have this argument anyways.

Differential Revision: [D16945279](https://our.internmc.facebook.com/intern/diff/D16945279)

Commit:
Add epsilon argument to Adagrad optimizer

bddppq commented on torch/optim/adagrad.py:
```diff
@@ -12,24 +12,28 @@ class Adagrad(Optimizer):
     """
 
-    def __init__(self, params, lr=1e-2, lr_decay=0, weight_decay=0, initial_accumulator_value=0):
+    def __init__(self, params, lr=1e-2, lr_decay=0, eps=1e-10, weight_decay=0, initial_accumulator_value=0):
```
better to put new arg to the end (for backward compatibility)

li-roy replied:
I ordered them this way initially to match the other optimizers, but yeah putting it at the end is better for bc.

bddppq reviewed (commented):
LGTM

Commit:
Update on "Add epsilon argument to Adagrad optimizer"

Add epsilon argument to Adagrad optimizer

gh-metadata: pytorch pytorch 24980 gh/li-roy/39/head

bddppq reviewed (approved):
LGTM

ezyang commented:
cc'ing Vincent who has been looking at optimizers a lot

ezyang commented:
@vincentqb You should consider adding yourself to module: optimizer on https://github.com/pytorch/pytorch/issues/24422
